### PR TITLE
proto: 1. add max_error_count for DAGRequest 2. add warning_count for SelectResponse

### PR DESCRIPTION
**Pull request by XuHuaiyu**
1. add `max_error_count` for `DAGRequest`
2. add `warning_count` for `SelectResponse`
PTAL @AndreMouche @coocood @hicqu

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

File: proto/select.proto [hunk 1]
```diff
@@ -80,6 +82,10 @@ message DAGRequest {
 
 	// It represents whether we collect the detailed scan counts in each range.
 	optional bool collect_range_counts = 6;
+
+	// It indicates the maximum number of error, warning, and note messages,
+	// and thus the number of messages that SHOW WARNINGS displays.
+	optional uint64 max_error_count = 7;
```
Comment (AndreMouche) [inline proto/select.proto]: It seems we only return one error now?

Comment (XuHuaiyu) [reply proto/select.proto]: @AndreMouche It does not only mean the count of error.
[This](https://dev.mysql.com/doc/refman/5.5/en/show-warnings.html) can be referred.

**Comment by XuHuaiyu** [comment]
PTAL @coocood

**Comment by zz-jason** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**